includes/scripts.php

```diff
@@ -34,6 +34,11 @@ function edd_load_scripts() {
 		$position = edd_get_item_position_in_cart( $post->ID );
 	}
 
+	$has_purchase_links = false;
+	if ( has_shortcode( $post->post_content, 'purchase_link' ) || has_shortcode( $post->post_content, 'downloads' ) || is_post_type_archive( 'download' ) ) {
```

Comment:
No the JS handles that part:
`( parseInt( edd_scripts.position_in_cart, 10 ) === parseInt( item, 10 ) )`.

Basically it forces the page to refresh if download we're viewing has a position in the cart already. The addition of this boolean check simply allows us to add another `edd_scripts` item for when we're not on a download that is currently in the cart.